### PR TITLE
Feat/adding login view

### LINKED ISSUE #13: Develop Main Login View  /  Desarrollar Vista Principal Del Login
**Explanation:**

  The main login view will be based on different sections. The image of DID Peru, message to enter DID Peru, the user field; which will be filled by the user who wishes to enter the application, password field, which will be filled in the same way by the user who wishes to enter the application. Also, enter the button. This button must be locked while the data entered is validated against the database. Besides; "Did you forget your password?" button. In addition to this, the "Associate with DID" button.

**User Stories to Fulfill:*

- [x] As a member, I want to place my username to be able to access my account.
- [x] As a member, I would like to enter a password to access my account.
- [ ] As a user, I would like to see the contact details of DID to acquire a membership.
- [ ] As a user, I would like to be able to reset my password.
- [ ] As a user, I want to be shown a support contact to reset my password.
- [x] As a user, I would like to see the "DID Peru" logo to be able to identify the app.
- [x] It has the message "Enter DID Peru"

### Dependencies:
It has no dependencies.

### References:
https://xd.adobe.com/view/a3457990-d0e6-4f82-876a-f382a7c73582-f616/screen/680ef50a-1656-43b7-ba65-e8b47775004c?hints=off

### Screenshot:
DID 2

### Workflow:
There is no workflow because it is the main screen.

Spanish Version:

**Explicación:**

La vista principal de acceso estará basada en diferentes secciones. La imagen de DID Perú, mensaje de ingresar a DID Perú, el campo de usuario; el cual será llenado por el usuario que desee ingresar a la aplicación, campo de contraseña, que será llenado de igual manera por el usuario que desea ingresar a la aplicación. También, el botón de ingresar. Este botón deberá ser bloqueado mientras se validan los datos ingresados con la base de datos. Así mismo; botón de "¿Olvidaste tu contraseña?". Adicional a esto, el botón de "Asociarme a DID".   

**Historias de Usuario a Cumplir:*

- [x]  Como miembro, deseo colocar mi usuario para poder acceder a mi cuenta.
- [x] Como miembro, me gustaría colocar una contraseña para acceder a mi cuenta.
- [ ] Como usuario, quisiera ver los datos de contacto de  DID para adquirir una membresía.
- [ ] Como usuario, me gustaría poder reestablecer mi contraseña.
- [ ] Como usuario, deseo que se me muestre un contacto de soporte para reestablecer mi contraseña. 
- [x] Como usuario, me gustaría ver el logo de "DID Perú" para poder identificar la app.
- [x] Cuenta con el mensaje de "Ingresa a DID Perú" 
 
### Dependencias:
No tiene dependencias.

### Referencias
https://xd.adobe.com/view/a3457990-d0e6-4f82-876a-f382a7c73582-f616/screen/680ef50a-1656-43b7-ba65-e8b47775004c?hints=off

### Captura de pantalla:
DID 2

### Flujo de trabajo:
No existe flujo de trabajo debido a que es pantalla principal.

### PR DESCRIPTION
**Pull request by heliomar-pena**
## Explanation

I've added the login view where the member could log in in her account. This is a change visual and too is preparing the infrastructure to create the login logic later

This issue fix some issues of the checkbox of #13 

- [x] Como miembro, quiero colocar mi usuario para poder acceder a mi cuenta.
- [x] Como miembro, quiero colocar una contraseña para acceder a mi cuenta.
- [x] Como usuario, quiero ver el logo de "DID Perú" para poder identificar la app.
- [x] Cuenta con el mensaje de "Ingresa a DID Perú" 

We still need to add on another PR:

- [ ] Como usuario, quiero ver datos de contacto de DID para adquirir una membresía. -> Need client information to be completed
- [ ] Como usuario, quiero poder reestablecer mi contraseña. -> Need client information to be completed 
- [ ] Como usuario, quiero que se me muestre un contacto de soporte para reestablecer mi contraseña. -> Need client information to be completed 

## Screenshots

image

## How to test?

1. Go to the link
2. Add the slug `/auth/login` at the end of the URL
3. Test in different screen sizes

### CODE REVIEW

File: src/actions/UserActions.ts [hunk 1]
```diff
@@ -1,19 +1,21 @@
...
-export const handleUserLogin = ({ email, password }: userAuthData) => {
+export const handleUserLogin = ({ user_code, password }: userAuthData) => {
   return new Promise((resolve, reject) => {
-    const variables: object = { email, password };
+    const variables: object = { user_code, password };
 
-    useQuery(getUserSessionQuery, variables)
+    useQuery<GetUsers>(getUserSessionQuery, variables)
       .then(({ users }) => {
+        console.log(users);
```
Comment (Israel-Laguan) [inline src/actions/UserActions.ts]: maybe remove this one

File: src/css/quasar.variables.scss [hunk 2]
```diff
@@ -12,13 +12,16 @@
 // to match your app's branding.
 // Tip: Use the "Theme Builder" on Quasar's documentation website.
 
-$primary   : #1976D2;
-$secondary : #26A69A;
-$accent    : #9C27B0;
+$primary: #1976d2;
+$secondary: #26a69a;
+$accent: #9c27b0;
```
Comment (Israel-Laguan) [inline src/css/quasar.variables.scss:17]: Is this because stylelint?

File: src/layouts/AuthLayout.vue [hunk 3]
```diff
@@ -0,0 +1,13 @@
+<template>
+  <Suspense>
+    <router-view />
+  </Suspense>
+</template>
+
+<script lang="ts">
```
Comment (Israel-Laguan) [inline src/layouts/AuthLayout.vue]: I do wonder if it count as TS if you don't use types on the file 😉  LOL

File: src/router/routes.ts [hunk 4]
```diff
@@ -14,7 +14,7 @@ const routes: RouteRecordRaw[] = [
   },
   {
     path: '/auth',
-    component: () => import('layouts/MainLayout.vue'),
+    component: () => import('src/layouts/AuthLayout.vue'),
```
Comment (Israel-Laguan) [inline src/router/routes.ts]: was 'src' really needed?

File: src/pages/Login/Login.layout.vue [hunk 5]
```diff
@@ -1,50 +1,72 @@
...
-      :hint="emailAttrs.hint"
-      :type="emailAttrs.type"
-      :model-value="emailValue"
-      @update:model-value="$emit('update:emailValue', $event)"
-      lazy-rules
-      :rules="emailValidations"
-    />
+  <div class="login__container">
+    <header>
+      <img class="login__logo" :src="didlogo" alt="DID Logo" />
```
Comment (Israel-Laguan) [inline src/pages/Login/Login.layout.vue]: maybe change the name of the image file to "didLogo"

File: src/css/quasar.variables.scss (hunk 2 above)
Comment (heliomar-pena) [reply src/css/quasar.variables.scss:17]: Yes, it's

File: src/layouts/AuthLayout.vue (hunk 3 above)
Comment (heliomar-pena) [reply src/layouts/AuthLayout.vue]: LOL. Would you suggest to remove `ts` if it's not used?

File: src/router/routes.ts (hunk 4 above)
Comment (heliomar-pena) [reply src/router/routes.ts]: I think it's not, I should remove it

File: src/pages/Login/Login.layout.vue (hunk 5 above)
Comment (heliomar-pena) [reply src/pages/Login/Login.layout.vue]: I'll do!

Comment (heliomar-pena) [reply src/pages/Login/Login.layout.vue]: I miss styled components hahaha

File: src/actions/UserActions.ts (hunk 1 above)
Comment (heliomar-pena) [reply src/actions/UserActions.ts]: Will do, thanks!

File: src/layouts/AuthLayout.vue (hunk 3 above)
Comment (Israel-Laguan) [reply src/layouts/AuthLayout.vue]: yeah

**Review by e-azocar — COMMENTED** [review]
Could you add validation for the "Código de socio" field?
image

**Comment by heliomar-pena** [comment]
> Could you add validation for the "Código de socio" field? image

Thanks for the suggestion ! @CARPER11 exists some validation for this field now? I remember some like "have 16 characters" but i'm not sure

**Comment by heliomar-pena** [comment]
> Could you add validation for the "Código de socio" field? image

Based on [this topic of rock](https://web.rock.so/space/4X20Boxp/topic/9) I've added a validation where the user code should have 10 character length and should be: "Two letters and eight numbers", for example: VP12345678

LastRecord

@CARPER11 We would need the messages to show to the user when an error occurs

File: src/utils/apollo.types.ts [hunk 6]
```diff
@@ -77,6 +77,8 @@ export interface product {
 }
 
 /** Query returns */
+// Products
```
Comment (Israel-Laguan) [inline src/utils/apollo.types.ts]: Maybe unify one-line comments style?

Comment (heliomar-pena) [reply src/utils/apollo.types.ts]: I thought to refactor this type's file, dividing it in different files (Products, Users, Categories, etc...), what do you think?

Comment (Israel-Laguan) [reply src/utils/apollo.types.ts]: Oh, I was referring that you have 2 styles:

`/**`
`//`

For one line comments. I think you should use `/*` only for multiline or JSDocs

Comment (heliomar-pena) [reply src/utils/apollo.types.ts]: Ok, sounds good!

**Comment by heliomar-pena** [comment]
I will merge this PR and continue improving the login view in their own spring. Thanks all for the code review

**Pull request merged**